File: hrms-backend/src/main/java/com/finalproject/hrmsbackend/business/abstracts/EmployerService.java
package com.finalproject.hrmsbackend.business.abstracts;

import com.finalproject.hrmsbackend.core.utilities.results.DataResult;
import com.finalproject.hrmsbackend.core.utilities.results.Result;
import com.finalproject.hrmsbackend.entities.concretes.Employer;
import com.finalproject.hrmsbackend.entities.concretes.dtos.EmployerAddDto;

import java.util.List;

public interface EmployerService {

    DataResult<Boolean> existsByCompanyName(String companyName);

    DataResult<Boolean> existsByWebsite(String website);

    DataResult<Employer> getById(int emplId);

    DataResult<Employer> getByEmailAndPW(String email, String password);

    DataResult<List<Employer>> getAll();

    DataResult<List<Employer>> getVerified();

    DataResult<List<Employer>> getUnverified();

    Result add(EmployerAddDto employerAddDto);

    Result updateCompanyName(String companyName, int emplId);

    Result updateEmailAndWebsite(String website, String email, int emplId);

    Result updatePhoneNumber(String phoneNumber, int emplId);

    Result applyChanges(int emplId);

    Result updateVerification(boolean systemVerificationStatus, int emplId);

}

File: hrms-frontend/src/services/employerService.js
import axios from "axios"

export default class EmployerService {

    existsByCompanyName(companyName) {
        return axios.get(`http://localhost:8080/api/employers/exists/byCompanyName?companyName=${companyName}`)
    }

    existsByWebsite(website) {
        return axios.get(`http://localhost:8080/api/employers/exists/byWebsite?website=${website}`)
    }

    getAll() {
        return axios.get("http://localhost:8080/api/employers/get/all");
    }

    getPublic() {
        return axios.get("http://localhost:8080/api/employers/get/public");
    }

    getById(emplId) {
        return axios.get(`http://localhost:8080/api/employers/get/byId?emplId=${emplId}`);
    }

    getByEmailAndPassword(email, password) {
        return axios.get(`http://localhost:8080/api/employers/get/byEmailAndPW?email=${email}&password=${password}`)
    }

    add(values) {
        return axios.post("http://localhost:8080/api/employers/add", values)
    }

    updateCompanyName(emplId, companyName) {
        return axios.put(`http://localhost:8080/api/employers/update/companyName?companyName=${companyName}&emplId=${emplId}`)
    }

    updateEmailAndWebsite(emplId, email, website) {
        return axios.put(`http://localhost:8080/api/employers/update/emailAndWebsite?email=${email}&emplId=${emplId}&website=${website}`)
    }

    updatePhoneNumber(emplId, phoneNumber) {
        return axios.put(`http://localhost:8080/api/employers/update/phoneNumber?emplId=${emplId}&phoneNumber=${phoneNumber}`)
    }

    applyChanges(emplId) {
        return axios.put(`http://localhost:8080/api/employers/update/applyChanges?emplId=${emplId}`)
    }

    updateVerification(id, status) {
        return axios.put(`http://localhost:8080/api/employers/update/verification?emplId=${id}&status=${status}`)
    }

}